ext/gmp/gmp.c

```diff
@@ -339,7 +339,24 @@ static zend_object *gmp_clone_obj(zend_object *obj) /* {{{ */
 /* }}} */
 
 static void shift_operator_helper(gmp_binary_ui_op_t op, zval *return_value, zval *op1, zval *op2, uint8_t opcode) {
-	zend_long shift = zval_get_long(op2);
+	bool failed = true;
+	zend_long shift = zval_try_get_long(op2, &failed);
+	if (failed) {
+		//char operator_sigil[2];
+		char *operator_sigil;
+		if (opcode == ZEND_POW) {
+			operator_sigil = "**";
+		} else if (opcode == ZEND_SL) {
+			operator_sigil = "<<";
+		} else {
+			ZEND_ASSERT(opcode == ZEND_SR);
+			operator_sigil = ">>";
+		}
+
+		zend_type_error("Unsupported operand types: GMP %s %s", operator_sigil, zend_zval_value_name(op2));
```

Comment:
`zval_try_get_long` can already throw. This will be chained, but is that desirable?

Reply:
For reasons that I don't understand it was not actually throwing an exception, might be doing something dumb

Reply:
Actually I double-checked, and it does not throw an exception (contrary to the non try version)

Reply:
If you start from an object then the function could fail with an exception. Similarly, passing a double will cause a conversion to a long which can warn and thus can also trigger an exception. So exceptions seem possible to me. I guess it's fine though to chain it with  the extra exception you're throwing here. Wdyt ?

Reply:
I created a test case for this:

```php
<?php

set_error_handler(function () {
    throw new Error('whoops');
});

$gmp = new GMP("5");
$gmp << 3.14;
```

Outputs this for me with this PR:
```
Fatal error: Uncaught Error: whoops in /run/media/niels/MoreData/php-src/x.php:4
Stack trace:
#0 /run/media/niels/MoreData/php-src/x.php(8): {closure:/run/media/niels/MoreData/php-src/x.php:3}(8192, 'Implicit conver...', '/run/media/niel...', 8)
#1 {main}

Next TypeError: Unsupported operand types: GMP << float in /run/media/niels/MoreData/php-src/x.php:8
Stack trace:
#0 {main}
  thrown in /run/media/niels/MoreData/php-src/x.php on line 8
```

maybe that's fine though.